pkg/controllers/provisioningv2/rke2/common.go

```diff
@@ -38,6 +38,7 @@ const (
 	MachineIDLabel             = "rke.cattle.io/machine-id"
 	MachineNameLabel           = "rke.cattle.io/machine-name"
 	MachineNamespaceLabel      = "rke.cattle.io/machine-namespace"
+	MachineOSAnnotation        = "rke.cattle.io/machine-os"
```

Comment:
Is there a reason we need this one and `CattleOSLabel`?

Reply:
So why create a new one and not continue to use the one that already exists?

Reply:
My thoughts behind making a new annotation and not using the label were as follows:
- rke1 and rke2 custom windows clusters use the `cattle.io/os` label for provisioning logic as the OS of the VM is known at the time of registering the node to a cluster
- node driver clusters do not have the `cattle.io/os` applied to them at any point during provisioning and I am unsure if it is currently added on after a machine is finished provisioning. 
- Since these are v2prov node driver pools and leveraging capi machines, I felt the use of an annotation matching the other naming structures of the v2prov labels/annotations was appropriate.

Additionally, this is meant to differentiate and make it easier to filter/search in a cluster for windows machines that are provisioned through v2prov.

Reply:
I am a fan of keeping this uniform in all these situations, but I am willing to be overruled by a majority.

In addition, we do use labels for somethings like roles.

Reply:
I will switch from using a new label/annotation to using CattleOSLabel